chanacceptor/zeroconfacceptor.go

```diff
@@ -0,0 +1,53 @@
+package chanacceptor
+
+// ZeroConfAcceptor wraps a regular ChainedAcceptor. If no acceptors are in the
+// ChainedAcceptor, then Accept will reject all channel open requests. This
+// should only be enabled when the zero-conf feature bit is set and is used to
+// protect users from a malicious counter-party double-spending the zero-conf
+// funding tx.
+type ZeroConfAcceptor struct {
+	chainedAcceptor *ChainedAcceptor
+}
+
+// NewZeroConfAcceptor initializes a ZeroConfAcceptor.
+func NewZeroConfAcceptor() *ZeroConfAcceptor {
+	return &ZeroConfAcceptor{
+		chainedAcceptor: NewChainedAcceptor(),
+	}
+}
+
+// AddAcceptor adds a sub-ChannelAcceptor to the internal ChainedAcceptor.
+func (z *ZeroConfAcceptor) AddAcceptor(acceptor ChannelAcceptor) uint64 {
+	return z.chainedAcceptor.AddAcceptor(acceptor)
+}
+
+// RemoveAcceptor removes a sub-ChannelAcceptor from the internal
+// ChainedAcceptor.
+func (z *ZeroConfAcceptor) RemoveAcceptor(id uint64) {
+	z.chainedAcceptor.RemoveAcceptor(id)
+}
+
+// Accept will deny the channel open request if the internal ChainedAcceptor is
+// empty. If the internal ChainedAcceptor has any acceptors, then Accept will
+// instead be called on it.
+//
+// NOTE: Part of the ChannelAcceptor interface.
+func (z *ZeroConfAcceptor) Accept(
+	req *ChannelAcceptRequest) *ChannelAcceptResponse {
+
+	if z.chainedAcceptor.numAcceptors() == 0 {
```

Comment:
Circling back to assert that this should only deny _iff_ this is a zero conf channel.